Makefile

```diff
@@ -1,13 +1,222 @@
+		}'
+
+.PHONY: targets
+targets:	## Display available targets and descriptions
+	@$(ECHO) "$(_HEADER)=== Makefile Targets [$(APP_NAME)] ===$(_NORMAL)"
+	@grep -v -E '.*(\?|\:\=).*$$' Makefile | grep -E '\#\#.*$$' \
+		| $(_SED) -r -e 's/([\sA-Za-z0-9\_\-]+)\:[\sA-Za-z0-9\_\-\.]+\s*\#{2,3}\s*(.*)\s*$$/\1!!!\2/g' \
+		| awk ' BEGIN {FS = "(:|###)+.*?## "}; \
+			/\#\#\#/ 	{printf "$(_SECTION)%s$(_NORMAL)\n", $$1;} \
+			/:/   	 	{printf "  $(_TARGET)%-$(TARGET_SIZE)s$(_NORMAL) %s\n", $$1, $$2;} \
+		  '
+	@$(ECHO) ""
+
+
+### Versioning Targets ###
+
+# Bumpversion 'dry' config
+# if 'dry' is specified as target, any bumpversion call using 'BUMP_XARGS' will not apply changes
+BUMP_XARGS ?= --verbose --allow-dirty	## Extra arguments to pass down to version control utility
+BUMP_XARGS := $(call clean_opt,$(BUMP_XARGS))
+ifeq ($(filter dry, $(MAKECMDGOALS)), dry)
+  BUMP_XARGS := $(BUMP_XARGS) --dry-run
+endif
+BUMP_CFG  ?= .bumpversion.cfg				## Bump version configuration (default recommended)
+BUMP_CFG  := $(call clean_opt,$(BUMP_CFG))
+BUMP_TOOL := bump2version
+BUMP_CMD  := $(BUMP_TOOL) --config-file "$(BUMP_CFG)"
+
+# guess the applicable semantic level update if provided via major|minor|patch targets
+# perform validation to avoid many combination provided simultaneously
+override BUMP_VERSION_INPUT := 1
+override BUMP_VERSION_LEVEL :=
+ifeq ($(filter major, $(MAKECMDGOALS)), major)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := major
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter minor, $(MAKECMDGOALS)), minor)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := minor
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifeq ($(filter patch, $(MAKECMDGOALS)), patch)
+  ifeq ($(BUMP_VERSION_LEVEL),)
+    override BUMP_VERSION_LEVEL := patch
+    override BUMP_VERSION_INPUT := 0
+  else
+    $(error "Cannot combine 'major|minor|patch' simultaneously!")
+  endif
+endif
+ifneq ($(VERSION),)
+  ifneq ($(BUMP_VERSION_LEVEL),)
+    $(error "Cannot combine 'major|minor|patch' simultaneously with explicit VERSION input!")
+  endif
+endif
+# when none was provided, use patch to directly apply the explicit 'VERSION' specified as input
+ifeq ($(BUMP_VERSION_LEVEL),)
+  override BUMP_VERSION_LEVEL := --new-version "$(VERSION)" patch
+endif
+
+.PHONY: dry
+dry: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: patch
+patch: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: minor
+minor: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+.PHONY: major
+major: $(BUMP_CFG)
+	@-$(ECHO) > /dev/null
+
+
+# run bumpversion with replacement of the release time value within its config for auto-update on future revisions
+.PHONY: bump
+bump: bump-check bump-install  ## Bump version using specified <VERSION> (call: make VERSION=<MAJOR.MINOR.PATCH> bump)
```

Comment:
Can we change the example here to include the usage: `make bump major|minor|patch`? I think I'm going to be using it with that usage 99% of the time.